pkg/actions/service.go

```diff
@@ -70,28 +82,87 @@ func (r *TaskResult) LogPath() string {
 	return FormatHookOutputPath(r.RunID, r.HookRunID)
 }
 
-type RunResultIterator interface {
-	Next() bool
-	Value() *RunResult
-	Err() error
-	Close()
+func runResultFromProto(pb *RunResultData) *RunResult {
+	return &RunResult{
+		RunID:     pb.RunId,
+		BranchID:  pb.BranchId,
+		SourceRef: pb.SourceRef,
+		EventType: pb.EventType,
+		CommitID:  pb.CommitId,
+		StartTime: pb.StartTime.AsTime(),
+		EndTime:   pb.EndTime.AsTime(),
+		Passed:    pb.Passed,
+	}
 }
 
-type TaskResultIterator interface {
-	Next() bool
-	Value() *TaskResult
-	Err() error
-	Close()
+func protoFromRunResult(m *RunResult) *RunResultData {
+	return &RunResultData{
+		RunId:     m.RunID,
+		BranchId:  m.BranchID,
+		CommitId:  m.CommitID,
+		SourceRef: m.SourceRef,
+		EventType: m.EventType,
+		StartTime: timestamppb.New(m.StartTime),
+		EndTime:   timestamppb.New(m.EndTime),
+		Passed:    m.Passed,
+	}
 }
 
-const defaultFetchSize = 1024
+func taskResultFromProto(pb *TaskResultData) *TaskResult {
+	return &TaskResult{
+		RunID:      pb.RunId,
+		HookRunID:  pb.HookRunId,
+		HookID:     pb.HookId,
+		ActionName: pb.ActionName,
+		StartTime:  pb.StartTime.AsTime(),
+		EndTime:    pb.EndTime.AsTime(),
+		Passed:     pb.Passed,
+	}
+}
 
-var ErrNotFound = errors.New("not found")
+func protoFromTaskResult(m *TaskResult) *TaskResultData {
+	return &TaskResultData{
+		RunId:      m.RunID,
+		HookRunId:  m.HookRunID,
+		HookId:     m.HookID,
+		ActionName: m.ActionName,
+		StartTime:  timestamppb.New(m.StartTime),
+		EndTime:    timestamppb.New(m.EndTime),
+		Passed:     m.Passed,
+	}
+}
+
+func getTasksPath(repoID, runID string) string {
+	return kv.FormatPath(actionsPrefix, repoID, taskPrefix, runID)
+}
+
+func getRunPath(repoID, runID string) string {
+	return kv.FormatPath(actionsPrefix, repoID, runPrefix, runID)
+}
+
+func getRunByBranchPath(repoID, branchID string) string {
+	return kv.FormatPath(actionsPrefix, repoID, branchPrefix, branchID)
+}
 
-func NewService(ctx context.Context, db db.Database, source Source, writer OutputWriter, stats stats.Collector, runHooks bool) *Service {
+func getRunByCommitPath(repoID, commitID string) string {
+	return kv.FormatPath(actionsPrefix, repoID, commitPrefix, commitID)
+}
```

Comment:
A suggestion to treat this as a REST API, longer but more readable paths. I think it would also help us expanding the API as we continue to improve the actions service.

1. Run path: 
	`actions/repos/<repoID>/runs/<runID>`
2. Tasks per Run path:
	`actions/repos/<repoID>/runs/<runID>/tasks/<hookRunID>`
3. Run by Branch path:
	`actions/repos/<repoID>/branches/<branchID>/runs/<runID>`
4. Run by Commit path:
	`actions/repos/<repoID>/commits/<commitID>/runs/<runID>`

Reply:
1-2: It's not a good idea to nest the paths of different data models - it will complicate scanning
3-4: Done